src/Nethermind/Nethermind.Merge.AuRa/AuRaMergeFinalizationManager.cs

```diff
@@ -0,0 +1,72 @@
+// SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using Nethermind.Blockchain;
+using Nethermind.Consensus;
+using Nethermind.Consensus.AuRa;
+using Nethermind.Consensus.Processing;
+using Nethermind.Core.Crypto;
+
+namespace Nethermind.Merge.Plugin
+{
+    public class AuRaMergeFinalizationManager : MergeFinalizationManager, IManualBlockFinalizationManager, IAuRaBlockFinalizationManager
+    {
+        private readonly IAuRaBlockFinalizationManager? _auRaBlockFinalizationManager;
+
+        public AuRaMergeFinalizationManager(IManualBlockFinalizationManager manualBlockFinalizationManager, IBlockFinalizationManager? blockFinalizationManager, IPoSSwitcher poSSwitcher) : base(manualBlockFinalizationManager, blockFinalizationManager, poSSwitcher)
+        {
+            _auRaBlockFinalizationManager = blockFinalizationManager as IAuRaBlockFinalizationManager;
```

Comment:
Does it make to instantiate this and put `blockFinalizationManager` as anything else than `IAuRaBlockFinalizationManager` ? If not why not make it explicit in constructor? And why use `as` instead of proper casting and throwing exceptions?

Reply:
refactored based on the suggestion